src/engine.ml

```diff
@@ -558,7 +571,34 @@ let kex_server config session (my_key_material : my_key_material) tls data =
       (Config.get Config.Data_ciphers config)
     |> Option.to_result ~none:(`Msg "No shared ciphers")
   in
+  let iv_proto =
+    let ( let* ) = Option.bind in
+    let prefix = "IV_PROTO=" in
+    let* peer_info = their_tls_data.peer_info in
+    let* iv_proto_s = List.find_opt (String.starts_with ~prefix) peer_info in
+    let v =
+      String.sub iv_proto_s (String.length prefix)
+        (String.length iv_proto_s - String.length prefix)
+    in
+    int_of_string_opt v
+  in
   let config = Config.add Cipher (cipher :> Config.cipher) config in
+  let config =
+    let supports_tls_ekm =
+      Option.fold iv_proto ~none:false
+        ~some:Packet.Iv_proto.(contains Tls_key_export)
+    in
+    let config =
+      if
+        supports_tls_ekm
+        && Option.fold iv_proto ~none:false
+             ~some:Packet.Iv_proto.(contains Use_cc_exit_notify)
+      then Config.add_protocol_flag `Tls_ekm config
+      else config
+    in
+    if supports_tls_ekm then Config.add Key_derivation `Tls_ekm config
+    else config
+  in
```

Comment:
After testing it seems OpenVPN **only** sends `protocol-flags tls-ekm` and no `key-derivation tls-ekm` if `Use_cc_exit_notify` bit is set. I guess it could make the code slightly simpler only to add either.